packages/runtime/container-runtime/src/orderedClientElection.ts

```diff
@@ -110,154 +136,173 @@ export class OrderedClientElection extends TypedEventEmitter<IOrderedClientElect
-        return this.summarizerCount;
+    /**
+     * Helper function to elect the first eligible client starting with the passed in client,
+     * or undefined if none are eligible.
+     * @param client - client to start checking
+     * @returns oldest eligible client starting with passed in client or undefined if none.
+     */
+    private electFirstEligibleClient(client: ILinkedClient | undefined): void {
+        let candidateClient = client;
+        while (candidateClient !== undefined && !this.isEligibleFn(candidateClient)) {
+            candidateClient = candidateClient.youngerClient;
+        }
+        this.tryElectingClient(candidateClient);
     }
 
-    /** Returns the total count of clients tracked. */
-    public getTotalCount(): number {
-        return this.clientMap.size;
+    /**
+     * Updates tracking for when a new client is added to the collection.
+     * Will automatically elect that new client if none is elected currently.
+     * @param client - client added to the collection
+     */
+    private addClient(client: ILinkedClient): void {
+        if (this.isEligibleFn(client)) {
+            this._eligibleCount++;
+            if (this._electedClient === undefined) {
+                // Automatically elect latest client
+                this.tryElectingClient(client);
+            }
+        }
     }
 
-    /** Returns an array of all eligible client ids being tracked in order from oldest to newest. */
-    public getOrderedEligibleClientIds(): string[] {
-        const result: string[] = [];
-        let currClient = this.rootClient;
-        while (currClient.youngerClient !== undefined) {
-            result.push(currClient.youngerClient.clientId);
-            currClient = currClient.youngerClient;
+    /**
+     * Updates tracking for when an existing client is removed from the collection.
+     * Will automatically elect next oldest client if currently elected is removed.
+     * @param client - client removed from the collection
+     */
+    private removeClient(client: ILinkedClient): void {
+        if (this.isEligibleFn(client)) {
+            this._eligibleCount--;
+            if (this._electedClient === client) {
+                // Automatically shift to next oldest client
+                this.electFirstEligibleClient(this._electedClient.youngerClient);
+            }
         }
-        return result;
     }
-}
 
-/**
- * Used to give increasing delay times for throttling a single functionality.
- * Delay is based on previous attempts within specified time window, ignoring actual delay time.
- */
- export class Throttler {
-    private startTimes: number[] = [];
-    constructor(
-        private readonly delayWindowMs: number,
-        private readonly maxDelayMs: number,
-        private readonly delayFunction: (n: number) => number,
-    ) { }
+    public getAllEligibleClients(): ITrackedClient[] {
+        return this.orderedClientCollection.getAllClients().filter(this.isEligibleFn);
+    }
```

Comment:
i wonder if it would be easier to just keep a map in this class as well as it will also remove the need for this._eligibleCount. i tend to dislike any kind of ref counting, as it's easy to break